flask_chat/views.py

```diff
@@ -0,0 +1,414 @@
+        logger.error("Unable to load user info.")
+        return render_template(
+            "error.html",
+            message="Unable to process info about your identity. Pleasy try later.",
+        )
+
+    logger.debug("Info about user retrieved successfully: %s", user_data)
+
+    return render_template(
+        "index.html",
+        msg_data=message_data,
+        usr_data=user_data.to_json(),
+        web_name=WEBSITE_NAME,
+    )
+
+
+# Registration page route
+@views_bp.route("/register", methods=["GET", "POST"])
+def register():
+    """Registration page"""
+
+    log_request()
+
+    # Signup form handle
+    form = RegForm(request.form)
+
+    if request.method != "POST":
+        return render_template("register.html", form=form)
+
+    # Here goes validating of WTForm inputs
+    if not form.validate():
+        logger.debug("Validation of registration form's input failed")
+        return render_template("register.html", form=form)
+
+    # If everything's ok, retrieve inputs from form
+    username = form.username.data
+    email = form.email.data
+    password = form.password.data
+
+    # Check whether user tries to register with used credentials
+    if user_service.get_user_info(username=username) or user_service.get_user_info(
+        email=email
+    ):
+        flash("These credentials are already in use.")
+        return redirect(request.url)
+
+    # Make sure insertion is completed without errors
+    result = user_service.insert_user(username, password, email)
+    if not result:
+        logger.error("New user registration failed")
+        return render_template(
+            "error.html",
+            message="Registration failed due to the yet unknown reasons. Try again later.",
+        )
+
+    logger.info("New user registration is completed successfully")
+
+    return redirect("/login")
+
+
+# Login page route
+@views_bp.route("/login", methods=["GET", "POST"])
+def login():
+    """Login page"""
+
+    log_request()
+
+    # Login form handle
+    form = LogForm(request.form)
+
+    if request.method != "POST":
+        return render_template("login.html", form=form)
+
+    if not form.validate():
+        logger.debug("Validation of registration form's input failed")
+        return render_template("login.html", form=form)
+
+    username = form.username.data
+    password = form.password.data
+
+    # Check whether user exists in the database
+    if not user_service.get_user_info(username=username):
+        flash("Invalid username or password.")
+        return render_template("login.html", form=form)
+
+    [user_data] = user_service.get_user_info(username=username)
+    if not user_data:
+        logger.error("Unable to load user info.")
+        return render_template(
+            "error.html",
+            message="Unable to process info about your identity. Pleasy try later.",
+        )
+
+    # Here goes verifying password hashes
+    if not user_data.verify_password(password):
```

Comment:
[CRITICAL] Please, combine this if with password verification. Validate what DB return if user not exists. If something wrong, return message should be the same all the time

Reply:
Fixed.